Middleware/ConnectionNameAwareInterface.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Middleware;
+
+interface ConnectionNameAwareInterface
+{
+    public function setConnectionName(string $name): void;
+}
```

Comment:
What's the use case for this?

Reply:
I wrote this PR with the replacement of `DebugStack` by middlewares in mind. In the symfony profiler, queries are grouped by connection. I faced a choice:
- either I inject in the middleware a different service (kind of debug data holder) by connection
- either the middleware is aware of the connection and set queries in an unique debug data holder tied with their connection
It's not a big deal to duplicate an abstract middle for each connection, but duplicating too the debug data holder seemed to me more complicated; it's why I chose the second possibility.

This is how I plan to create the debug data holder:
```php
class DebugDataHolder
{
    private array $data = [];

    public function addQuery(string $connexionName, Query $query): void
    {
        $this->data[$connexionName][] = [
            'sql' => $query->getSql(),
            'params' => $query->getParams(),
            'types' => $query->getTypes(),
            'executionMS' => static fn() => $query->getDuration(),  // stop() may not be called at this point
        ];
    }

    public function getData(string $connexionName): array
    {
        if (!isset($this->data[$connexionName])) {
            return [];
        }

        foreach ($this->data[$connexionName] as &$data) {
            if (is_callable($data['executionMS'])) {
                $data['executionMS'] = $data['executionMS']();
            }
        }

        return $this->data[$connexionName];
    }
}
```